### PR TITLE
Updated functions.cpp and Added Tests

### PR DESCRIPTION
**Pull request by ShikharJ**

### CODE REVIEW

**Comment by ShikharJ** [comment]
ping @isuruf.

File: symengine/functions.cpp [hunk 1]
```diff
@@ -1584,8 +1584,7 @@ bool Csch::is_canonical(const RCP<const Basic> &arg) const
 RCP<const Basic> csch(const RCP<const Basic> &arg)
 {
     if (eq(*arg, *zero)) {
-        // Answer is infinity. Yet to be implemented in SymEngine
-        throw NotImplementedError("Not Implemented");
+        return infty(0);
```
Comment (isuruf) [inline symengine/functions.cpp]: Use `return ComplexInf;`

File: symengine/tests/basic/test_functions.cpp [hunk 2]
```diff
@@ -2480,6 +2487,9 @@ TEST_CASE("Gamma: functions", "[functions]")
     r2 = one;
     REQUIRE(eq(*r1, *r2));
 
+    r1 = gamma(minus_one);
+    REQUIRE(r1->__str__() == "zoo");
```
Comment (isuruf) [inline symengine/tests/basic/test_functions.cpp]: Use `eq(*r1, *ComplexInf)` instead of comparing using strings.

**Comment by isuruf** [comment]
@ShikharJ, if it's just one job that needs to be restarted, let me know. I can just restart that job instead of the whole build which has 20 jobs.

**Pull request merged**